runner/sidecar/sources.go

```diff
@@ -22,6 +21,39 @@ import (
 )
 
 func connectSources(ctx context.Context, process func(context.Context, []byte) error) error {
+	var pendingGauge *prometheus.GaugeVec
```

Comment:
looks good to me - do you plan to remove these from the step.status in a follow-up PR?

Reply:
yes.

Reply:
ok - I approve - but you need to fix test before merge